### PR TITLE
SIM-1712 ADD documentation for prepared statement rule

### PR DESCRIPTION
**Pull request by arditymeri**

### CODE REVIEW

File: docs/rules/replace-dynamic-query-by-prepared-statement.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+:::
+
+## Code Changes
+
+
+### Using Statement execute
+
+__Pre__
+```java
+String departmentId = "40 OR '1'='1";
```
Comment (gregor-balint) [inline docs/rules/replace-dynamic-query-by-prepared-statement.md:50]: I would prefer a simple initialization containing a valid number, for example:
```java
String departmentId = "'40'";

File: docs/rules/replace-dynamic-query-by-prepared-statement.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+String query = "SELECT first_name FROM employee WHERE department_id ='" + departmentId + "' ORDER BY last_name";
+Statement statement = connection.createStatement();
+statement.execute(query);
+ResultSet resultSet = statement.getResultSet();
+```
+
+__Post__
+```java
+String departmentId = "40 OR '1'='1";
```
Comment (gregor-balint) [inline docs/rules/replace-dynamic-query-by-prepared-statement.md:59]: see above, applies for all similar initializations below

File: docs/rules/replace-dynamic-query-by-prepared-statement.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+int id = 10;
+String query = "SELECT first_name FROM employee WHERE  id >  ?" + "";
+query += " AND department_id ='" + departmentId + "' ORDER BY last_name";
+PreparedStatement statement = connection.prepareStatement(query);
+statement.setInt(1, id);
+ResultSet resultSet = statement.executeQuery();
+```
+
```
Comment (gregor-balint) [inline docs/rules/replace-dynamic-query-by-prepared-statement.md:107]: I would add a simple negative example without parameter:
```java
 String query = "SELECT first_name FROM employee WHERE department_id ='40' ORDER BY last_name";
```

File: docs/rules/replace-dynamic-query-by-prepared-statement.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+Statement statement = connection.createStatement();
+ResultSet resultSet = statement.executeQuery(query);
+```
+
+__Post__
+```java
+String departmentId = "40 OR '1'='1";
+int id = 10;
+String query = "SELECT first_name FROM employee WHERE  id >  ?" + "";
+query += " AND department_id ='" + departmentId + "' ORDER BY last_name";
```
Comment (gregor-balint) [inline docs/rules/replace-dynamic-query-by-prepared-statement.md:102]: Interesting idea: maybe it is possible to replace also the second parameter after the '+=' concatenation in 
```java
query += " AND department_id ='" + departmentId + "' ORDER BY last_name";
```
for example by recognizing that the statement beginning with  'query +=' immediately follows the initialization statement of the query.

File: docs/rules/replace-dynamic-query-by-prepared-statement.md (hunk 1 above)
Comment (arditymeri) [reply docs/rules/replace-dynamic-query-by-prepared-statement.md:50]: The idea behind "40 OR '1'" is to make it clear where the threat of SQL injection is coming from

File: docs/rules/replace-dynamic-query-by-prepared-statement.md (hunk 4 above)
Comment (arditymeri) [reply docs/rules/replace-dynamic-query-by-prepared-statement.md:102]: yes yes. I have that in mind. I will fix that. Thanks for pointing it out again.

**Pull request merged**